### ISSUE TITLE
Memedroid

### ISSUE THREAD

**Issue by Heus-Sueh** [yt-dlp/yt-dlp#11150]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Brazil, EUA, French, Italy, Spain, Portugal and others.

### Example URLs

- Single Video: https://www.memedroid.com/memes/detail/4373699/OMW-No-Homo

### Provide a description that is worded well enough to be understood

To extract the video from the website it seems easy, just search for the `source` with the name `detailed-item-video-src-mp4` in the html:

```html
<source id="detailed-item-video-src-mp4" src="https://rvideos1.memedroid.com/videos/UPLOADED920/66fdc31a31e66.mp4" type="video/mp4">
```

- Source file in the HTML: https://rvideos2.memedroid.com/videos/UPLOADED854/66fd685e93c5b.mp4

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
🞈 yt-dlp -vU https://www.memedroid.com/memes/detail/4373699/OMW-No-Homo
[debug] Command-line config: ['-vU', 'https://www.memedroid.com/memes/detail/4373699/OMW-No-Homo']
[debug] User config "/var/home/heus/.config/yt-dlp/config": ['--ignore-errors', '-o', '~/Videos/%(title)s.%(ext)s', '-f', 'bestvideo[ext=mp4][width<2000][height<=1200]+bestaudio[ext=m4a]/bestvideo[ext=webm][width<2000][height<=1200]+bestaudio[ext=webm]/bestvideo[width<2000][height<=1200]+bestaudio/best[width<2000][height<=1200]/best']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.09.27 from yt-dlp/yt-dlp [c6387abc1] (pip)
[debug] Python 3.12.6 (CPython x86_64 64bit) - Linux-6.10.11-200.fc40.x86_64-x86_64-with-glibc2.39 (OpenSSL 3.0.14 4 Jun 2024, glibc 2.39)
[debug] exe versions: ffmpeg 6.1.2 (setts), ffprobe 6.1.2
[debug] Optional libraries: Cryptodome-3.21.0, brotli-1.1.0, certifi-2024.08.30, mutagen-1.47.0, requests-2.32.3, sqlite3-3.46.0, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1838 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.09.27 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.09.27 from yt-dlp/yt-dlp)
[generic] Extracting URL: https://www.memedroid.com/memes/detail/4373699/OMW-No-Homo
[generic] OMW-No-Homo: Downloading webpage
ERROR: [generic] Unable to download webpage: HTTP Error 403: Forbidden (caused by <HTTPError 403: Forbidden>)
  File "/var/home/heus/.local/pipx/venvs/yt-dlp/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 741, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/heus/.local/pipx/venvs/yt-dlp/lib/python3.12/site-packages/yt_dlp/extractor/generic.py", line 2384, in _real_extract
    full_response = self._request_webpage(url, video_id, headers=filter_dict({
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/heus/.local/pipx/venvs/yt-dlp/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 910, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/var/home/heus/.local/pipx/venvs/yt-dlp/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 897, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query, extensions))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/heus/.local/pipx/venvs/yt-dlp/lib/python3.12/site-packages/yt_dlp/YoutubeDL.py", line 4165, in urlopen
    return self._request_director.send(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/heus/.local/pipx/venvs/yt-dlp/lib/python3.12/site-packages/yt_dlp/networking/common.py", line 117, in send
    response = handler.send(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/heus/.local/pipx/venvs/yt-dlp/lib/python3.12/site-packages/yt_dlp/networking/_helper.py", line 208, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/home/heus/.local/pipx/venvs/yt-dlp/lib/python3.12/site-packages/yt_dlp/networking/common.py", line 340, in send
    return self._send(request)
           ^^^^^^^^^^^^^^^^^^^
  File "/var/home/heus/.local/pipx/venvs/yt-dlp/lib/python3.12/site-packages/yt_dlp/networking/_requests.py", line 365, in _send
    raise HTTPError(res, redirect_loop=max_redirects_exceeded)
yt_dlp.networking.exceptions.HTTPError: HTTP Error 403: Forbidden
```

**Comment by dirkf** [comment]
Same result upstream, even though the page looks like plain HTML5 video that should be handled. It works fine in the browser even with minimal request headers.

**Comment by seproDev** [comment]
This seems to just be a Cloudflare 403. Using the `--impersonate` option, downloading works fine with the generic extractor:
```
yt-dlp "https://www.memedroid.com/memes/detail/4373699/OMW-No-Homo" --impersonate chrome -v
[debug] Command-line config: ['https://www.memedroid.com/memes/detail/4373699/OMW-No-Homo', '--impersonate', 'chrome', '-v']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version master@2024.10.01.001408 from yt-dlp/yt-dlp-master-builds [e59c82a74] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22631-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 7.0.2-full_build-www.gyan.dev (setts), ffprobe 7.0.2-full_build-www.gyan.dev, phantomjs 2.5.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.3, websockets-13.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1838 extractors
[generic] Extracting URL: https://www.memedroid.com/memes/detail/4373699/OMW-No-Homo
[generic] OMW-No-Homo: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] OMW-No-Homo: Extracting information
[debug] Looking for embeds
[debug] Identified a html5 embed
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] OMW-No-Homo-1: Downloading 1 format(s): 1
[debug] Invoking http downloader on "https://rvideos1.memedroid.com/videos/UPLOADED920/66fdc31a31e66.mp4"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: OMW, No Homo (1) [OMW-No-Homo-1].mp4
[download] 100% of  609.77KiB in 00:00:00 at 3.23MiB/s
```
You will have to install curl_cffi for this.

**Comment by dirkf** [comment]
The [cookies work-around](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#user-content-im-getting-http-error-403-and-the-site-has-an-open-issue-on-the-tracker-thats-labeled-cloudflare-related-what-can-i-do), in case _curl_cffi_ is not suitable or unwanted, doesn't seem to work, here or upstream.

A patch would only be possible if anyone knew a (=another) way to bypass CF fingerprint blocking.

**Comment by bashonly** [comment]
This will be fixed by #11206 (which enables impersonation by default for the generic extractor), unless there is additional metadata that is both available and desired from the site
